examples/react/basic/src/gen/eliza-ElizaService_connectquery.ts

```diff
@@ -96,24 +91,34 @@ export const ElizaService = {
   },
 } as const;
 
-const $queryService = createQueryService({ service: ElizaService });
-
 /**
  * Say is a unary RPC. Eliza responds to the prompt with a single sentence.
  *
  * @generated from rpc connectrpc.eliza.v1.ElizaService.Say
  */
-export const say: UnaryFunctionsWithHooks<SayRequest, SayResponse> = {
-  ...$queryService.say,
-  ...createUnaryHooks($queryService.say),
-};
+export const say = {
+  localName: "say",
+  name: "Say",
+  kind: MethodKind.Unary,
+  I: SayRequest,
+  O: SayResponse,
+  service: {
+    typeName: "connectrpc.eliza.v1.ElizaService",
+  },
+} as const;
 
 /**
  * SayAgain is a unary RPC. Eliza responds to the prompt with a single sentence.
  *
  * @generated from rpc connectrpc.eliza.v1.ElizaService.SayAgain
  */
-export const sayAgain: UnaryFunctionsWithHooks<SayRequest, SayResponse> = {
-  ...$queryService.sayAgain,
-  ...createUnaryHooks($queryService.sayAgain),
-};
+export const sayAgain = {
+  localName: "sayAgain",
+  name: "SayAgain",
+  kind: MethodKind.Unary,
+  I: SayRequest,
+  O: SayResponse,
+  service: {
+    typeName: "connectrpc.eliza.v1.ElizaService",
+  },
+} as const;
```

Comment:
We generate this line further up:

```ts
export const typeName = "connectrpc.eliza.v1.ElizaService";
```

We should either use this const, or remove it. 

If we want to use the const, we have to avoid name conflicts. Currently, an `rpc TypeName(TypeNameRequest) returns (TypeNameResponse)` will generate invalid code.

On a separate note: We generate the full `ElizaService` above. Ultimately, it would be great to have a self-describing type for methods as a first-class primitive, and have ConnectRouter.rpc() accept it, for example:

```diff
const t = createRouterTransport(router => {
-  router.rpc(say.service, say, (req) => {
+  router.rpc(say, (req) => {
    return new SayResponse({sentence: `you said: ${req.sentence}`});
  })
});
```

If we can settle on the shape, I'm in favor of modifying the router transport to accept this type as well.

Reply:
Yes, absolutely. If we're happy with the method declaration of `MethodUnaryDescriptor`, then we can propagate that back into `ConnectRouter` and I can delete this extra stuff, which would be great.

Reply:
I can submit a PR to `@bufbuild/protobuf` and `@connectrpc/connect` to contain whatever those represent. The only question I have is the use of `localName`, I'm not sure it needs to exist in the definition, since it really can be generated if anyone needs it. Thoughts @srikrsna-buf ?

Reply:
Went ahead and implemented it here: https://github.com/bufbuild/protobuf-es/pull/594. I'm all ears for what we'd want to name it though, `MethodType` isn't my favourite but it felt consistent with `ServiceType`.

Reply:
👍  let's discuss it over there.

Reply:
This was discussed in https://github.com/bufbuild/protobuf-es/pull/594, and implemented as an overload for `ConnectRouter.rpc` in https://github.com/connectrpc/connect-es/pull/884. Not released yet, so the code cannot be updated yet.